wasm/section_test.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2018 The go-interpreter Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package wasm_test
+
+import (
+	"bytes"
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"github.com/go-interpreter/wagon/wasm"
+)
+
+func TestSectionCustom(t *testing.T) {
+	fname := "testdata/custom_funcs_locals.wasm"
+
+	t.Run(filepath.Base(fname), func(t *testing.T) {
+		raw, err := ioutil.ReadFile(fname)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		r := bytes.NewReader(raw)
+		m, err := wasm.DecodeModule(r)
+		if err != nil {
+			t.Fatalf("error reading module %v", err)
+		}
+
+		nameCustom := m.Custom("name")
+		if nameCustom == nil {
+			t.Fatal("can not find name custom section")
+		}
+
+		var nSec wasm.NameSection
+		err = nSec.UnmarshalWASM(bytes.NewReader(nameCustom.Data))
+		if err != nil {
+			t.Fatalf("error name Section Unmarshal %v", err)
+		}
+
+		if len(nSec.Types[wasm.NameFunction]) > 0 {
```

Comment:
please invert the boolean test:
```suggestion
		if len(nSec.Types[wasm.NameFunction]) == 0 {
		    t.Fatalf(...)
		}
```